Expose missing Quaternion operators.

27thLiz opened the pull request:
Scripts can now evaluate the following cases:
- (quat \* real) and (quat / real)
- (quat + quat) and (quat - quat)

akien-mga commented on core/variant_op.cpp:
```diff
@@ -477,7 +477,13 @@ void Variant::evaluate(const Operator& p_op, const Variant& p_a, const Variant&
 					DEFAULT_OP_LOCALMEM(+,VECTOR3,Vector3);
 					DEFAULT_OP_FAIL(PLANE);
-					DEFAULT_OP_FAIL(QUAT);
+					case QUAT: {
```
Can't it be declared the same way as for `Vector3`, using `DEFAULT_OP_LOCALMEM(+,QUAD,Quad);`?

27thLiz committed:
Expose missing Quaternion operators.

Scripts can now evaluate the following cases:
 - (quat * real) and (quat / real)
 - (quat + quat) and (quat - quat)

Hinsbart commented:
> Can't it be declared the same way as for Vector3, using DEFAULT_OP_LOCALMEM(+,QUAT,Quat);?

Right, rebased.

Hinsbart commented:
Mhh, CI failed on the android build, but it doesn't look like it was the code's fault ^^

neikeq commented:
Restarted android build

akien-mga commented on core/variant_op.cpp:
```diff
@@ -699,7 +702,13 @@ void Variant::evaluate(const Operator& p_op, const Variant& p_a, const Variant&
 				DEFAULT_OP_LOCALMEM_NUM(/,VECTOR3,Vector3);
 				DEFAULT_OP_FAIL(PLANE);
-				DEFAULT_OP_FAIL(QUAT);
+				case QUAT: {
```
No `DEFAULT_OP_LOCALMEM_NUM(/, QUAT, Quat)` for this operation? (Just asking, I have no clue how this macro works :p)

Hinsbart replied:
`DEFAULT_OP_LOCALMEM_NUM` not only provides the operation on numbers but also on the type itself. 
However, Quat / Quat is not a valid op.

akien-mga replied:
Thanks for the clarification :)
